cookbook/templates/default/agent.yml.erb

```diff
@@ -0,0 +1,104 @@
+server:
+  type: simple
+  applicationContextPath: <%= node[:baragon][:agent_app_context_path] %>
+  connector:
+    type: http
+    port: <%= node[:baragon][:agent_port] %>
+
+zookeeper:
+  quorum: <%= node[:baragon][:zk_hosts].join(',') %>  # comma separated list of ZK host:port goes here
+  zkNamespace: <%= node[:baragon][:zk_namespace] %>    # base zk node goes here
+  sessionTimeoutMillis: 60000
+  connectTimeoutMillis: 5000
+  retryBaseSleepTimeMilliseconds: 1000
+  retryMaxTries: 3
```

Comment:
Seems like all of these should be attributes too…

I've built something like this recently by having a large hash (e.g., `node[:baragon][:agent][:config]`) and converting that by way of Ruby into the template format that's needed. This allows for the software Chef is managing to change without needing to “chase” the software’s config values in the recipe/templates.